docs/intro/overview.rst

```diff
@@ -19,143 +19,221 @@ so you can get an idea of how it works and decide if Scrapy is what you need.
+  spider responses do no affect other callbacks.
+
+Let's illustrate these points by building simple
+self-contained spider. Let's say we want to extract the URL, name and 
+number of forks for trending Python repositories in Github. 
+The list of trending repos can be found here: https://github.com/trending?l=python
+
+We can see that this page displays name, url and number of stars for 
+each repo, but it does not show number of forks. To get number of
+forks we will have to make request to each repository page and extract
+data from there. 
 
-Let's say we want to extract the URL, name, description and size of all torrent
-files added today in the `Mininova`_ site.
+.. _intro-overview-item:
 
-The list of all torrents added today can be found on this page:
+We need to write a Spider which visits some initial url
+(in our case https://github.com/trending?l=python), extracts links, 
+follows them by issuing GET requests and extracts data from repository pages.
 
-    http://www.mininova.org/today
+.. highlight:: html
 
-.. _intro-overview-item:
+Extract and follow links
+========================
 
-Define the data you want to scrape
-==================================
+Let's start with link extraction. If we open our browser developer tools 
+and look at page source we'll see that all URLs leading to repositories 
+are wrapped in html that looks like this::
 
-The first thing is to define the data we want to scrape. In Scrapy, this is
-done through :ref:`Scrapy Items <topics-items>` (Torrent files, in this case).
+    <h3 class="repo-list-name">
+        <a href="/honnibal/spaCy">
+        .....
+        </a> 
+    </h3>
 
-This would be our Item::
+.. highlight:: none
 
-    import scrapy
+We'll use `XPath`_ for selecting proper links to repositories. In our case the
+xpath will look like this::
 
-    class TorrentItem(scrapy.Item):
-        url = scrapy.Field()
-        name = scrapy.Field()
-        description = scrapy.Field()
-        size = scrapy.Field()
+    //h3[@class='repo-list-name']/a/@href
 
-Write a Spider to extract the data
-==================================
+.. highlight:: python
 
-The next thing is to write a Spider which defines the start URL
-(http://www.mininova.org/today), the rules for following links and the rules
-for extracting the data from pages.
+To create a Spider that is going to process initial url, extract and follow
+linkes that we are interested in we need to write following piece of code::
 
-If we take a look at that page content we'll see that all torrent URLs are like
-``http://www.mininova.org/tor/NUMBER`` where ``NUMBER`` is an integer. We'll use
-that to construct the regular expression for the links to follow: ``/tor/\d+``.
+    from urlparse import urljoin
 
-We'll use `XPath`_ for selecting the data to extract from the web page HTML
-source. Let's take one of those torrent pages:
+    from scrapy import Spider
+    from scrapy.http import Request
```

Comment:
Yeah, I think it is better - we use "import scrapy" consistently in the docs. We discussed it when implementing the shortcuts (can't find a link now) and agreed to use "import scrapy" and then scrapy.Item, scrapy.Spider, scrapy.Request, etc. consistently in the documentation examples and in the generated templates.